Sources/PackageGraph/PinsStore.swift

```diff
@@ -117,7 +110,7 @@ public final class PinsStore {
     public func createConstraints() -> [RepositoryPackageConstraint] {
         return pins.map({ pin in
             return RepositoryPackageConstraint(
-                container: pin.packageRef, requirement: pin.state.requirement(), products: pin.productFilter)
+                container: pin.packageRef, requirement: pin.state.requirement(), products: .everything)
```

Comment:
Won’t this constraint mean that every pin could potentially want _more of its transitive dependencies_ after it’s loaded than when it was pinned?

If a package is resolved was at 1.0.0 for use of product B, then it will be pinned at 1.0.0 with no product information (which ought to be somehow inferable). Upon loading the pins, it will be loaded at 1.0.0 for use of products A and B, and the resolver will go hunting for all of product A’s dependencies. Or is that not what happens?

Passing `.specific([])` might work instead, because I _think_ the resolver will still get the remaining product‐based constraints from the manifests anyway.

I will try both variants and get back to you.

Reply:
These constraints are no longer actually being used by the resolver, I removed this method in #2793

Reply:
Yes, I see that now. It makes things much simpler.